### PR TITLE
Test dropping union fields more

### PR DESCRIPTION
**Pull request by syvb**
Now that #87403 is merged, a few more tests can be added for reads/writes to dropping union fields.

r? @LeSeulArtichaut

### CODE REVIEW

File: src/test/ui/unsafe/union-assignop.rs [hunk 1]
```diff
@@ -0,0 +1,30 @@
...
+    b: Dropping, // treated as dropping
+}
+
+fn main() {
+    let mut foo = Foo { a: 42 };
+    foo.a += 5; //~ ERROR access to union field is unsafe
+    foo.b += Dropping; //~ ERROR access to union field is unsafe
+    foo.b = Dropping; //~ ERROR assignment to union field that might need dropping is unsafe
+    foo.a; //~ ERROR access to union field is unsafe
+    let foo = Foo { a: 42 };
```
Comment (LeSeulArtichaut) [inline src/test/ui/unsafe/union-assignop.rs:24]: Why do you need to re-assign to `foo` here?

**Review by LeSeulArtichaut — APPROVED** [review]
:+1: Having more tests for this is definitely a good idea. Just one small thing and we can merge.

File: src/test/ui/unsafe/union-assignop.rs (hunk 1 above)
Comment (Smittyvb) [reply src/test/ui/unsafe/union-assignop.rs:24]: Otherwise the borrow checker complains that `foo` is used after being moved out of.

**Comment by LeSeulArtichaut** [comment]
Thanks for the PR! @bors r+ rollup

**Pull request merged**